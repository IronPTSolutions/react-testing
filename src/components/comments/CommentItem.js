import React from 'react'

const CommentItem = (props) => (
  <div className="card">
    <div className="card-header d-flex justify-content-between align-items-center" data-testid="user">
      {props.user} 
      <button className="btn btn-link text-danger" data-testid="delete-btn"onClick={props.onClickDelete}><i className="fa fa-times"></i></button>
    </div>
    <div className="card-body">
      <p data-testid="message">{props.message}</p>
    </div>
    <div className="card-footer text-muted" data-testid="created-at">{props.createdAt.toDateString()}</div>
  </div>
);

export default CommentItem;