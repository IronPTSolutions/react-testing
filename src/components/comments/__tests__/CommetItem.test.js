import React from 'react'
import { cleanup, render, fireEvent } from '@testing-library/react'
import 'jest-dom/extend-expect'
import CommentItem from '../CommentItem';

afterEach(cleanup)

it('should show props at dom properly', () => {
  const component = render(<CommentItem user="john.doe@example.org" createdAt={new Date('2019-07-19')} message={"test message"}/>)
  expect(component.getByTestId('user')).toHaveTextContent('john.doe@example.org')
  expect(component.getByTestId('created-at')).toHaveTextContent('Fri Jul 19 2019')
  expect(component.getByTestId('message')).toHaveTextContent('test message')
})


it('should call onDeleteClick prop properly on click in delete comment button', () => {
  const clickSpy = jest.fn()
  const component = render(<CommentItem user="john.doe@example.org" createdAt={new Date('2019-07-19')} message={"test message"} onClickDelete={clickSpy}/>)
  fireEvent.click(component.getByTestId('delete-btn'))
  expect(clickSpy).toHaveBeenCalledTimes(1)
})

it('should render as snapshot', () => {
  const clickSpy = jest.fn()
  const component = render(<CommentItem user="john.doe@example.org" createdAt={new Date('2019-07-19')} message={"test message"} onClickDelete={clickSpy} />)
  expect(component.asFragment()).toMatchSnapshot()
})