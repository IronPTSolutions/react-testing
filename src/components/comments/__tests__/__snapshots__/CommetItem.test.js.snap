// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render as snapshot 1`] = `
<DocumentFragment>
  <div
    class="card"
  >
    <div
      class="card-header d-flex justify-content-between align-items-center"
      data-testid="user"
    >
      john.doe@example.org
      <button
        class="btn btn-link text-danger"
        data-testid="delete-btn"
      >
        <i
          class="fa fa-times"
        />
      </button>
    </div>
    <div
      class="card-body"
    >
      <p
        data-testid="message"
      >
        test message
      </p>
    </div>
    <div
      class="card-footer text-muted"
      data-testid="created-at"
    >
      Fri Jul 19 2019
    </div>
  </div>
</DocumentFragment>
`;
